pkg/machines/xmlCreator.es6

```diff
@@ -12,6 +12,7 @@ export function getDiskXML(poolName, volumeName, target) {
 
     var targetElem = doc.createElement('target');
     targetElem.setAttribute('dev', target);
+    targetElem.setAttribute('bus', 'virtio');
```

Comment:
Libvirt will not choose what is best according to the OS or architecture or whatever (because that's considered a policy and libvirt doesn't do policy decisions). Libvirt chooses the disk bus type for new disks according to the target dev that we pass. In cockpit we were and are building the disk target devices with the `vd` prefix, so vda, vdb, etc.
https://github.com/cockpit-project/cockpit/blob/master/pkg/machines/components/diskAdd.jsx#L35..L47
In this way we are telling libvirt indirectly that we want virtio bus. That's what is happening when a user is creating a new disk from the cockpit UI to attach it to a existing domain.

So with this change we basically change nothing for now, I just wanted to have it explicit to avoid confusion. (I should probably clear it out in the commit message)

In the future, I would like to move xml creation to virt-xml so we 'll stop making this kind of decisions and let virt-xml which will have libosinfo knowledge do the configuration for the disks.

The situation when creating new domains is a bit different. For that we use ```virt-install``` and ```virt-install``` has is own logic to find the next available target and the bus type for the disks. So we don't make any decision there, and  it can choose ide bus type, we don't restrict it. That BZ you mention was for this case.